src/common/undo.c

```diff
@@ -225,6 +226,8 @@ void dt_undo_do_redo(dt_undo_t *self, uint32_t filter)
     l = g_list_next(l);
   }
   UNLOCK;
+
+  dt_collection_update_query(darktable.collection);
```

Comment:
I suppose this could be done only for lighttable undo types (that is no doing that when history or mask is changed), something like that:

if(filter & DT_UNDO_LIGHTTABLE)  dt_collection_update_query(darktable.collection);

Reply:
We need this to work on the filmstrip, on discard history, when an image is first edited, etc. Actually any operation that may change the collection.

Reply:
I see, and indeed I forgot that a collection could be changed by history stack if the aspect ratio is changed. So looks good indeed.